codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/S3ErrorWith200Status.java

```diff
@@ -0,0 +1,72 @@
+package software.amazon.smithy.aws.go.codegen.customization;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import software.amazon.smithy.aws.traits.ServiceTrait;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.go.codegen.GoDelegator;
+import software.amazon.smithy.go.codegen.GoSettings;
+import software.amazon.smithy.go.codegen.GoWriter;
+import software.amazon.smithy.go.codegen.SymbolUtils;
+import software.amazon.smithy.go.codegen.integration.GoIntegration;
+import software.amazon.smithy.go.codegen.integration.MiddlewareRegistrar;
+import software.amazon.smithy.go.codegen.integration.RuntimeClientPlugin;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.OperationShape;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.utils.ListUtils;
+
+/**
+ * Adds middleware to handle S3 response errors with 200 ok status code.
+ */
+public class S3ErrorWith200Status implements GoIntegration {
+    private static String ADD_ERROR_HANDLER_INTERNAL = "HandleResponseErrorWith200Status";
+
+    // list of operations for which this customization is valid.
+    private static Set<String> customizedOperations = Stream.of(
+            "CopyObject", "UploadPartCopy", "CompleteMultipartUpload")
+            .collect(Collectors.toCollection(HashSet::new));
+
+
+    @Override
+    public byte getOrder() {
+        // The associated customization ordering is relative to operation deserializers
+        // and thus the integration should be added at the end.
+        return 127;
+    }
+
+    @Override
+    public List<RuntimeClientPlugin> getClientPlugins() {
+        return ListUtils.of(
+                RuntimeClientPlugin.builder()
+                        .operationPredicate(S3ErrorWith200Status::supports200Error)
+                        .registerMiddleware(MiddlewareRegistrar.builder()
+                                .resolvedFunction(SymbolUtils.createValueSymbolBuilder(ADD_ERROR_HANDLER_INTERNAL,
+                                        AwsCustomGoDependency.S3_CUSTOMIZATION).build())
+                                .build())
+                        .build()
+        );
+    }
+
+    // returns true if the operation supports error response with 200 ok status code
+    private static boolean supports200Error(Model model, ServiceShape service, OperationShape operation){
+        if (!isS3Service(model, service)) {
+            return false;
+        }
+
+        if (customizedOperations.contains(operation.getId().getName())) {
+            return true;
+        }
+        return false;
```

Comment:
You can just return the result of contains

Reply:
Right, I missed refactoring this. Thanks